src/CMakeLists.txt

```diff
@@ -54,6 +54,14 @@ set(DEP_LIBS_PKG_ARG_LISTS "IpcShm ${DEP_IPC_SHM_VERSION} CONFIG")
 # (TODO: but look into the jemalloc/jemalloc-cmake GitHub repo -- what's that all about?)
 # Hence we must ensure 2 things more manually: the lib location and the include-path.
 
+# Try pkg-config if available
+pkg_check_modules(JEMALLOC REQUIRED jemalloc)
+if(JEMALLOC_FOUND)
+  set(JEMALLOC_LIB "${JEMALLOC_LIBRARIES}")
+  set(JEMALLOC_INCLUDE_PATH "${JEMALLOC_INCLUDEDIR}")
+  unset(JEMALLOC_PREFIX CACHE)
```

Comment:
Is this necessary? If they specified this prefix manually for some unholy reason (and we do advertise this), maybe we should let them? (I have the vague feeling CMake's philosophy on cache variables isn't totally clear... at least I couldn't glean it when I tried... like this value could be a wrongly-computed value from a previous run, in which case who knows what we should do - but presumably worst-case user can just clear cache and start over....)

Reply:
I meant just the unset() line there... not the 3 preceding that. Apologies for being confusing.

Reply:
https://cmake.org/cmake/help/latest/module/FindPkgConfig.html#command:pkg_check_modules says that a `<YYY>_PREFIX` is set by `pkg_check_modules()`, and I was trying to avoid potential conflict. (I mistakenly thought `JEMALLOC_PREFIX` is only set by the following `jemalloc-config` utilities and not supposed to be specified by users.)

Reply:
Okay, got it. There's quite a bit going on. I can surely improve this based on stuff you've taught me (plus subsequent experiments). Like I don't need to introduce more cache variables if some would already be set with the pkg... stuff... and clearly I should differentiate in naming between prefix as in path prefix, versus prefix as in the API name prefix (which is super duper important here). And probably more boring subtleties like that.

The time to fix it up is now, before (god willing) all kinds of users start using mr Flow-IPC. 

So, not ready to go yet, but I'm optimistic. 👍🏻

Reply:
A different name like `JEMALLOC_API_PREFIX` is inconsistent with upstream `jemalloc`'s convention, which might be confusing.

I think some "save/restore" actions might help here, with proper explaining comments: Save user specified `JEMALLOC_PREFIX` to another variable temporarily and restore `JEMALLOC_PREFIX` with that saver variable after `pkg_check_modules()` stuffs.

Be bold and do it! Changes are never too late. 👍🏻👍🏻👍🏻